polyfill/lib/datetime.mjs

```diff
@@ -216,9 +339,21 @@ export class DateTime {
       microseconds,
       nanoseconds
     ));
-    days -= deltaDays;
-    ({ year, month, day } = ES.SubtractDate(year, month, day, years, months, weeks, days, disambiguation));
-    ({ year, month, day, hour, minute, second, millisecond, microsecond, nanosecond } = ES.RegulateDateTime(
+
+    // Delegate the date part subtraction to the calendar
+    const calendar = GetSlot(this, CALENDAR);
```

Comment:
Why do you do the calendar arithmetic before time arithmetic in plus, but after time arithmetic in minus?